sdk/dotnet/Pulumi/Deployment/GrpcMonitor.cs

```diff
@@ -0,0 +1,39 @@
+using System.Threading.Tasks;
+using Grpc.Core;
+using Pulumirpc;
+
+namespace Pulumi
+{
+    class GrpcMonitor : IMonitor
+    {
+        private readonly ResourceMonitor.ResourceMonitorClient _client;
+        private readonly Engine.EngineClient _engine;
+
+        public GrpcMonitor(string engine, string monitor)
+        {
+            this._engine = new Engine.EngineClient(new Channel(engine, ChannelCredentials.Insecure));
+            this._client = new ResourceMonitor.ResourceMonitorClient(new Channel(monitor, ChannelCredentials.Insecure));
+        }
+
+        public async Task<InvokeResponse> InvokeAsync(InvokeRequest request)
+            => await this._client.InvokeAsync(request);
+
+        public async Task<ReadResourceResponse> ReadResourceAsync(Resource resource, ReadResourceRequest request)
+            => await this._client.ReadResourceAsync(request);
+
+        public async Task<RegisterResourceResponse> RegisterResourceAsync(Resource resource, RegisterResourceRequest request)
+            => await this._client.RegisterResourceAsync(request);
+
+        public async Task RegisterResourceOutputsAsync(RegisterResourceOutputsRequest request)
+            => await this._client.RegisterResourceOutputsAsync(request);
+
+        public async Task LogAsync(LogRequest request)
+            => await this._engine.LogAsync(request);
+
+        public async Task<SetRootResourceResponse> SetRootResourceAsync(SetRootResourceRequest request)
+            => await this._engine.SetRootResourceAsync(request);
+
+        public async Task<GetRootResourceResponse> GetRootResourceAsync(GetRootResourceRequest request)
+            => await this._engine.GetRootResourceAsync(request);
```

Comment:
can remove async/await from all of these methods.

Reply:
ah, missed this when got this code from IDE auto-refactoring. any idea why they do it by default?

Reply:
which refactoring were you doing?

Reply:
It's actually more tricky: if I remove async-await, I get
`Cannot convert expression type 'Grpc.Core.AsyncUnaryCall<Pulumirpc.InvokeResponse>' to return type 'System.Threading.Tasks.Task<Pulumirpc.InvokeResponse>'`
So, `await` "converts" `AsyncUnaryCall` to `Task`.